examples/default/variables.tf

```diff
@@ -16,6 +16,11 @@ variable "prefix" {
   default     = "test-landing-zone"
 }
 
+variable "is_vpc_existing" {
+  description = "Indicates whether user wants to use an existing vpc or create a new one."
+  type        = bool
+}
+
```

Comment:
Based partially on the COS module as example.

This support almost certainly requires two new properties (to avoid the terraform feature about using values created at apply time in counts). It likely also requires an addition example to validate using existing values.

For variables to be consistent with the COS module (there may be other conflicting examples you would prefer to follow)

existing_cos_instance_id and create_cos_instance become existing_vpc_id (string) and create_vpc (bool), extending to existing_subnet_ids (map(string)) and create_subnets (bool)... etc.